dask/dataframe/tests/test_dataframe.py

```diff
@@ -1155,22 +1155,23 @@ def test_repartition_on_pandas_dataframe():
 
 
 def test_repartition_npartitions():
-    df = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6], 'y': list('abdabd')},
-                      index=[10, 20, 30, 40, 50, 60])
-    for n in [1, 2, 4, 5]:
-        for k in [1, 2, 4, 5]:
-            if k > n:
-                continue
-            a = dd.from_pandas(df, npartitions=n)
-            k = min(a.npartitions, k)
-
-            b = a.repartition(npartitions=k)
-            eq(a, b)
-            assert b.npartitions == k
-
-    a = dd.from_pandas(df, npartitions=1)
-    with pytest.raises(ValueError):
-        a.repartition(npartitions=5)
+    for index in (None, [10, 20, 30, 40, 50, 60]):
+        df = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6], 'y': list('abdabd')},
+                          index=index)
+        for n in [1, 2, 4, 5]:
+            for k in [1, 2, 4, 5]:
+                if k > n:
+                    continue
+                a = dd.from_pandas(df, npartitions=n)
+                k = min(a.npartitions, k)
+
+                b = a.repartition(npartitions=k)
+                eq(a, b)
+                assert b.npartitions == k
+
+        a = dd.from_pandas(df, npartitions=1)
+        with pytest.raises(ValueError):
+            a.repartition(npartitions=5)
```

Comment:
I'm not sure that this test tests what you intend.  Passing `index=None` to `pd.DataFrame` just sends the index `[0, 1, 2, ...]`.  I suspect you'll get a previously-failing-test if you pass `sort=False` to `dd.from_pandas`.